### PR TITLE
executor: support Chunk for TableScanExec

### PR DESCRIPTION
**Pull request by zz-jason**
Here we will convert `[]Datum` to `Chunk` during the execution of `NextChunk`, after we supporting `Chunk` for `table.Table` interface, we can remove the conversion. 

to #5261

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -795,16 +795,112 @@ type TableScanExec struct {
...
+	}
+
+	for chk.NumRows() < e.ctx.GetSessionVars().MaxChunkSize {
+		row, err := e.getRow(handle)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		e.seekHandle = handle + 1
+		for i, col := range e.Schema().Columns {
+			appendDatum2Chunk(chk, &row[i], i, col.RetType)
```
Comment (coocood) [inline executor/executor.go]: consider use `chunk.MutRow`.
1. set Datums to `MutRow`
2. Append `MutRow` to chunk.

File: executor/executor.go [hunk 2]
```diff
@@ -795,16 +795,113 @@ type TableScanExec struct {
...
+func (e *TableScanExec) nextChunk4InfoSchema(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.virtualTableChunkList == nil {
+		e.virtualTableChunkList = chunk.NewList(e.Schema().GetTypes(), e.ctx.GetSessionVars().MaxChunkSize)
+		columns := make([]*table.Column, e.schema.Len())
+		for i, colInfo := range e.columns {
+			columns[i] = table.ToColumn(colInfo)
+		}
+		virtualTableChunk := e.newChunk()
+		var mutableRow chunk.MutRow
```
Comment (coocood) [inline executor/executor.go]: It need to be initialized by types.

File: executor/executor.go [hunk 3]
```diff
@@ -795,16 +795,112 @@ type TableScanExec struct {
...
+		mutableRow.SetDatums(row...)
+		chk.AppendRow(0, mutableRow.ToRow())
+	}
+	return nil
+}
+
+func (e *TableScanExec) nextChunk4InfoSchema(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.virtualTableChunkList == nil {
+		e.virtualTableChunkList = chunk.NewList(e.Schema().GetTypes(), e.ctx.GetSessionVars().MaxChunkSize)
```
Comment (coocood) [inline executor/executor.go]: e.maxChunkSize

**Comment by shenli** [comment]
@zz-jason Please fix CI.

File: executor/executor.go [hunk 4]
```diff
@@ -795,16 +795,112 @@ type TableScanExec struct {
...
+	if e.virtualTableChunkIdx >= e.virtualTableChunkList.NumChunks() {
+		return nil
+	}
+	virtualTableChunk := e.virtualTableChunkList.GetChunk(e.virtualTableChunkIdx)
+	e.virtualTableChunkIdx++
+	chk.SwapColumns(virtualTableChunk)
+	return nil
+}
+
+func (e *TableScanExec) nextHandle() (handle int64, found bool, err error) {
```
Comment (shenli) [inline executor/executor.go]: Add comment for this function.

File: executor/executor.go [hunk 5]
```diff
@@ -795,16 +795,112 @@ type TableScanExec struct {
...
+		e.virtualTableChunkList = chunk.NewList(e.Schema().GetTypes(), e.maxChunkSize)
+		columns := make([]*table.Column, e.schema.Len())
+		for i, colInfo := range e.columns {
+			columns[i] = table.ToColumn(colInfo)
+		}
+		virtualTableChunk := e.newChunk()
+		mutableRow := chunk.MutRowFromTypes(e.Schema().GetTypes())
+		err := e.t.IterRecords(e.ctx, nil, columns, func(h int64, rec []types.Datum, cols []*table.Column) (bool, error) {
+			mutableRow.SetDatums(rec...)
+			virtualTableChunk.AppendRow(0, mutableRow.ToRow())
```
Comment (coocood) [inline executor/executor.go]: We can just append the row to `virtualTableChunkList` directly.

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@shenli PTAL

File: executor/executor.go [hunk 6]
```diff
@@ -898,14 +957,11 @@ func (e *TableScanExec) getRow(handle int64) (Row, error) {
 // Open implements the Executor Open interface.
 func (e *TableScanExec) Open(goCtx goctx.Context) error {
 	e.iter = nil
+	e.virtualTableChunkList = nil
```
Comment (shenli) [inline executor/executor.go]: e. virtualTableRows = nil

File: plan/logical_plan_builder.go [hunk 7]
```diff
@@ -1640,12 +1640,15 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
 		}
 	}
 	ds.SetSchema(schema)
-	if handleCol == nil {
+	isMemDB := infoschema.IsMemoryDB(ds.DBName.L)
+	if handleCol == nil && !isMemDB {
```
Comment (shenli) [inline plan/logical_plan_builder.go]: Add comment here.

**Comment by shenli** [comment]
LGTM

**Pull request merged**